packages/eas-build-job/src/generic.ts

```diff
@@ -39,4 +35,17 @@ export namespace Generic {
     triggeredBy: z.literal(BuildTrigger.GIT_BASED_INTEGRATION),
     loggerLevel: z.nativeEnum(LoggerLevel).optional(),
   });
+
+  export const JobZ = z.union([
+    CommonJobZ.extend({
+      customBuildConfig: z.object({
```

Comment:
Interesting, for some reason I have been imagining we would put `steps` _under_ `customBuildConfig`. It does make sense to keep it out though. Cool!